invenio_rdm_records/fixtures/vocabularies.py

```diff
@@ -105,20 +108,157 @@ def __iter__(self):
             yield from create_iterator(data_file)
 
 
+#
+# Exceptions
+#
+class ConflictingFixturesError(Exception):
+    """Exception when multiple modules provide same subject."""
+
+    def __init__(self, errors):
+        """Constructor."""
+        message = "\n".join(errors)
+        super().__init__(message)
+        self.errors = errors
+
+
 #
 # Fixture
 #
+class VocabulariesFixtures:
```

Comment:
naming: to be honest it is quite confusing to have `VocabulariesFixtures` and `VocabulariesFixture`. It makes it quite hard to discern what's the different - I'm actually not sure I understand 😅 

- The first one allows to load (potentially) from external packages. being the priority `app_data`, `packages` then `rdm-records` (e.g. `data`)
- The second one loads as before: first `app_data` then `data`

In addition, `VocabulariesFixtures` uses `VocabularyFixture` to load from `data` at the end. But the loading from `app_data` seems a bit duplicated since we need to put the packages/extensions in the middle of the loading right?

I'm not sure if this would be sensible, and I *do not want* to make you refactor a lot... But what about just 1 class that takes a "loading order" and has components to do so... Somehow similar to what we do in services with the `components`... So a `PackageFixture`, `AppDataFixture`, etc. and then there is just an order and they have a common interface. In some sort of "common" structure we can do the duplication check by passing it to the current loading component.

That said, maybe I totally misunderstood the code... hope not 😅

Reply:
The idea was to re-use the existing code as much as possible because we are only adding another level on top :sweat_smile: . tldr; I think changing `VocabulariesFixture._search_paths/_filename` to `filepath` would be a sensible change that diffuses some of the complexity since `VocabulariesFixture` really only represents a single `vocabularies.yaml` file and is the same functionality as each of the `PackageFixture`, `AppDataFixture` mentioned above.

Here is what we are trying to do:

Load fixtures in general (Vocabularies fixtures and others like users fixtures). Especially for vocabularies it means: loading vocabularies fixtures in the priority order app_data, extensions, data/ (package) BUT allowing for this priority to be at the subject type level rather than the vocabulary level for subjects (or any vocabulary with subtypes). It's that wrinkle that makes things a little strange I think. We need to revisit for the use case of updating an instance with a new voabulary anyway though.

Vocabularies fixtures are very layered:
 - there is the vocabuaries.yaml file with vocabularies in it
 - each vocabulary entry in that file has 1 or more data-files
 - each datafile has 0 or more records
so having a dedicated class for them is sensible.

Reply:
I see... It is indeed complex. I will try to understand the code more in depth... I find it a bit complex, but it is a complex task 🤷‍♂️ . I dont want to block you, so maybe just a more descriptive docstring to differentiate between `VocabulariesFixture` and `VocabulariesFixtures` 👼